Newsletters: add 184 (2022-01-26) (#712)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-01-26-newsletter.md

  without the cooperation of the receiver.  In short, the spender can
  prove to a court or the public what they paid for.

    However, in order for the receiver to subsequently construct a
    signature spending their received funds, they need the hash of the
    contract in addition to the key they used (that key usually being
    part of an [HD keychain][topic bip32]).  Orlovsky's proposal allows
    that hash to be added to a PSBT so that a signing wallet or hardware
    device can produce a valid signature.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [Is it possible to convert a taproot address into a native segwit address?]({{bse}}111440)

>   After an exchange changed a user's segwit v1 P2TR taproot withdrawal address
>   into a segwit v0 native segwit P2WSH address due to lack of taproot support,

Reviewer: Both P2TR and P2WSH are native segwit addresses. Also, "segwit v0 native segwit" feels a bit off. Perhaps this would be better:

```suggestion
  After an exchange changed a user's P2TR (native segwit v1) taproot withdrawal address
  into a P2WSH (native segwit v0) address due to lack of taproot support,
```

  the user asks if there is a way to claim the bitcoins in the resulting v0
  output. Pieter Wuille notes that those bitcoins are not retrievable since the
  user would need to find a script that hashes to the public key in the P2TR
  address, a computationally infeasible operation.

- [Was Bitcoin 0.3.7 actually a hard fork?]({{bse}}111673)
  User BA20D731B5806B1D wonders what about Bitcoin's 0.3.7 release caused it to
  be classified as a hard fork. Antoine Poinsot gives example `scriptPubKey`
  and `scriptSig` values to illustrate that a previously invalid signature could
  be valid after [0.3.7][bitcoin 0.3.7 github]'s bugfix to separate `scriptSig` +
  `scriptPubKey` evaluations.

- [What is signature grinding?]({{bse}}111660)
  Murch explains that ECDSA signature grinding is the process of repeatedly
  signing until you get a signature whose r-value is in the lower half of the
  range, resulting in a signature that is 1 byte smaller (32 bytes vs 33 bytes)

>   based on ECDSA's serialization format. This smaller signature results in less

Reviewer 2: ```suggestion
  based on ECDSA's serialization format. This smaller signature results in lower
```

  structure][wiki message structure], in order to identify if they are
  connected to a peer that is on the same network (mainnet, testnet, signet).

- [How many BIPs were adopted in the standard client in 2021?]({{bse}}111901)
  Pieter Wuille links to Bitcoin Core's [BIPs documentation][bitcoin bips doc]
  that keeps track of BIPs that are implemented in Bitcoin Core.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

<!-- FIXME:harding update on Tuesday -->

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Eclair #2134][] enables [anchor outputs][topic anchor outputs] by default,

>   allowing commitment transactions to be fee bumped should its feerate be too

Reviewer: Plural subject with singular possessive pronoun. Perhaps one of these:

```suggestion
  allowing a commitment transaction to be fee bumped should its feerate be too
```

```suggestion
  allowing commitment transactions to be fee bumped should their feerate be too
```

  low at broadcast time. Since anchor outputs-style fee bumping works via
  [CPFP][topic cpfp], users will need to keep UTXOs available in their
  `bitcoind` wallet.

- [Eclair #2113][] adds automatic management of fee bumping.  This
  includes classifying transactions by the importance of confirming them
  on time, re-evaluating transactions after each block to determine
  whether it's appropriate to fee bump them, also re-evaluating current

>   network feerates to in case the transaction's feerate needs to be

Reviewer: Methinks that "to" is a stray

```suggestion
  network feerates in case the transaction's feerate needs to be
```

  increased, and adding additional inputs to transactions if necessary
  to increase the transaction's feerate.  The PR description also makes
  an appeal for [improvements][Bitcoin Core #23201] to Bitcoin Core's
  wallet API that could reduce the amount of add-on wallet management
  required by external programs like Eclair.

- [Eclair #2133][] begins relaying [onion messages][topic onion
  messages] by default.  The rate limits mentioned in [Newsletter
  #181][news181 onion] are used to prevent problems from any abuse of
  this experimental part of the LN protocol.

- [BTCPay Server #3083][] allows administrators to log into a BTCPay
  instance using [LNURL authentication][] (which is also possible to

>   implement in non-LN software).

Reviewer: Perhaps:
```suggestion
 be implemented in non-LN software).
```
